objectiveC-day7.txt

```diff
@@ -0,0 +1,195 @@
+
+1)
+Counting the number of places where the object reference is used is called reference counting.
+When an object is created, the initial reference count is set to 1,Each time you need to ensure that the object be kept around, you effectively create a
+reference to it by incrementing its reference count by 1.
+
+Before Arc was introduced,programmers used 2 methods.
+a) automatic garbage collection:
+
+    With garbage collection, the system automatically determines which objects own which
+    other objects, automatically freeing up (or garbage-collecting) objects that are no longer referenced as space is needed during the program’s execution.
```

Comment:
There is no garbage collector is present in iOS. iOS uses ARC, before ARC manual referencing for memory management.